python/ray/gateway.py

```diff
@@ -0,0 +1,110 @@
+import click
+from flask import Flask, request, send_file
+import io
+import pickle
+import pyarrow
+from pyarrow import plasma as plasma
+import sys
+
+app = Flask(__name__)
+ctx = pyarrow.default_serialization_context()
+
+@app.route('/', methods=['GET', 'POST'])
+def index():
+    """TODO (dsuo): Add comments.
+
+    TODO (dsuo): Maybe care about batching
+    TODO (dsuo): REST is nice conceptually, but probably too much overhead
+    TODO (dsuo): Move logic to C++ (keep as head node process)
+
+    NOTE: We do an extra serialize during get and extra deserialize during
+        put.
+    """
+    if request.method == 'POST':
+        object_id = request.files['object_id'].read()
+        data = request.files['value'].read()
+        value = ctx.deserialize(data)
+
+        plasma_client.put(
+            value,
+            object_id=plasma.ObjectID(object_id),
+            memcopy_threads=12,
+            serialization_context=ctx)
+
+        return object_id, 402
+    elif request.method == 'GET' and 'object_ids' in request.args:
+        object_ids = [plasma.ObjectID(bytes.fromhex(object_id)) for object_id in \
```

Comment:
`bytes.fromhex` doesn't appear to exist in Python 2.

We have some utilities like `ray.utils.hex_to_binary`.

Reply:
Thanks for pointer!